### ISSUE TITLE
BugSnag crash when started in Application.attachBaseContext()

### ISSUE THREAD

**Issue by baolongnt** [bugsnag/bugsnag-android#952]
### Describe the bug
App crashes when BugSnag.start() is called in Application.attachBaseContext() instead of Application.onCreate().
We want to start bug reporting then to catch issue in content provider init.

```
2020-10-13 09:35:25.923 13799-13799/com.nguyentrong.bugsnagattachbasecontextcrash E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.nguyentrong.bugsnagattachbasecontextcrash, PID: 13799
    java.lang.RuntimeException: Unable to instantiate application com.nguyentrong.bugsnagattachbasecontextcrash.App: java.lang.IllegalArgumentException: Parameter specified as non-null is null: method kotlin.jvm.internal.Intrinsics.checkParameterIsNotNull, parameter context
        at android.app.LoadedApk.makeApplication(LoadedApk.java:1226)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6431)
        at android.app.ActivityThread.access$1300(ActivityThread.java:219)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1859)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
     Caused by: java.lang.IllegalArgumentException: Parameter specified as non-null is null: method kotlin.jvm.internal.Intrinsics.checkParameterIsNotNull, parameter context
        at com.bugsnag.android.ConnectivityCompat.<init>(Unknown Source:2)
        at com.bugsnag.android.Client.<init>(Client.java:119)
        at com.bugsnag.android.Bugsnag.start(Bugsnag.java:65)
        at com.bugsnag.android.Bugsnag.start(Bugsnag.java:41)
        at com.nguyentrong.bugsnagattachbasecontextcrash.App.attachBaseContext(App.kt:11)
        at android.app.Application.attach(Application.java:351)
        at android.app.Instrumentation.newApplication(Instrumentation.java:1149)
        at android.app.LoadedApk.makeApplication(LoadedApk.java:1218)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6431) 
        at android.app.ActivityThread.access$1300(ActivityThread.java:219) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1859) 
        at android.os.Handler.dispatchMessage(Handler.java:107) 
        at android.os.Looper.loop(Looper.java:214) 
        at android.app.ActivityThread.main(ActivityThread.java:7356) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930) 
```

### Steps to reproduce
Just start the app

### Environment
* Android version: 10
* Bugsnag version: 5.2.0
* Emulator or physical device: emulator

### Example Repo
https://github.com/baolongnt/BugSnagAttachBaseContextCrash

**Comment by baolongnt** [comment]
Something like this should fix the issue

```diff
diff --git a/bugsnag-android-core/src/main/java/com/bugsnag/android/Client.java b/bugsnag-android-core/src/main/java/com/bugsnag/android/Client.java
index bdb00da6..dde9573c 100644
--- a/bugsnag-android-core/src/main/java/com/bugsnag/android/Client.java
+++ b/bugsnag-android-core/src/main/java/com/bugsnag/android/Client.java
@@ -115,7 +115,9 @@ public class Client implements MetadataAware, CallbackAware, UserAware {
      * @param configuration  a configuration for the Client
      */
     public Client(@NonNull Context androidContext, @NonNull final Configuration configuration) {
-        appContext = androidContext.getApplicationContext();
+        appContext = androidContext instanceof Application
+                     ? androidContext
+                     : androidContext.getApplicationContext();
 
         connectivity = new ConnectivityCompat(appContext, new Function2<Boolean, String, Unit>() {
             @Override
```

**Comment by mattdyoung** [comment]
Hi @baolongnt 

Thanks for the report and for the suggested solution.

We plan to schedule releasing a fix for this in the near future.

**Comment by phillipsam** [comment]
Hi @baolongnt we have now released a fix for this in version [v5.2.2](https://github.com/bugsnag/bugsnag-android/releases/tag/v5.2.2).
